src/http/server/request_processor.cr

```diff
@@ -69,4 +86,16 @@ class HTTP::Server::RequestProcessor
       end
     end
   end
+
+  # Reads and discards bytes from `io` until there are no more bytes.
+  # If there are more than 16_384 bytes to be read from the IO, it returns `false`.
+  private def skip_to_end(io : IO) : Bool
+    buffer = uninitialized UInt8[4096]
+
+    4.times do
+      return true if io.read(buffer.to_slice) < 4096
```

Comment:
`read` isn't guaranteed to read the full slice. You should use `read_fully?`

Reply:
I don't think this part should be merged anyway. It's better to be strict and don't reuse the connection when there is even only one unconsumed byte.